l1-contracts/deploy-scripts/RegisterCTM.s.sol

```diff
@@ -0,0 +1,290 @@
+import {L1NullifierDev} from "contracts/dev-contracts/L1NullifierDev.sol";
+import {DiamondProxy} from "contracts/state-transition/chain-deps/DiamondProxy.sol";
+import {BridgedStandardERC20} from "contracts/bridge/BridgedStandardERC20.sol";
+import {ChainRegistrar} from "contracts/chain-registrar/ChainRegistrar.sol";
+import {ValidiumL1DAValidator} from "contracts/state-transition/data-availability/ValidiumL1DAValidator.sol";
+import {RollupDAManager} from "contracts/state-transition/data-availability/RollupDAManager.sol";
+import {BytecodesSupplier} from "contracts/upgrades/BytecodesSupplier.sol";
+import {L2LegacySharedBridgeTestHelper} from "./L2LegacySharedBridgeTestHelper.sol";
+import {ChainAdminOwnable} from "contracts/governance/ChainAdminOwnable.sol";
+import {ServerNotifier} from "contracts/governance/ServerNotifier.sol";
+import {UpgradeStageValidator} from "contracts/upgrades/UpgradeStageValidator.sol";
+import {L2AssetRouter} from "contracts/bridge/asset-router/L2AssetRouter.sol";
+import {CTMDeploymentTracker} from "contracts/bridgehub/CTMDeploymentTracker.sol";
+
+import {DeployUtils, GeneratedData, Config, DeployedAddresses} from "./DeployUtils.s.sol";
+import {ContractsBytecodesLib} from "./ContractsBytecodesLib.sol";
+
+contract RegisterCTM is Script, DeployUtils {
+    using stdToml for string;
+
+    struct Output {
+        address governance;
+        bytes encodedData;
+    }
+
+    function run() public virtual {
+        console.log("Registering CTM");
+
+        runInner(
+            "/script-config/config-deploy-l1.toml",
+            "/script-out/output-deploy-l1.toml",
+            "/script-out/register-ctm-l1.toml",
+            true
+        );
+    }
+
+    function registerCTM(bool shouldSend) public virtual {
+        console.log("Registering CTM");
+
+        runInner(
+            "/script-config/config-deploy-l1.toml",
+            "/script-out/output-deploy-l1.toml",
+            "/script-out/register-ctm-l1.toml",
+            shouldSend
+        );
+    }
+
+    function runForTest() public {
+        runInnerForTest(vm.envString("L1_CONFIG"), vm.envString("L1_OUTPUT"));
+    }
+
+    function getAddresses() public view returns (DeployedAddresses memory) {
+        return addresses;
+    }
+
+    function getConfig() public view returns (Config memory) {
+        return config;
+    }
+
+    function runInner(
+        string memory inputPath,
+        string memory inputPathIfEcosystemDeployedLocally,
+        string memory outputPath,
+        bool shouldSend
+    ) internal {
+        string memory root = vm.projectRoot();
+        inputPath = string.concat(root, inputPath);
+        inputPathIfEcosystemDeployedLocally = string.concat(root, inputPathIfEcosystemDeployedLocally);
+
+        initializeConfig(inputPath);
+        initializeConfigIfEcosystemDeployedLocally(inputPathIfEcosystemDeployedLocally);
+
+        registerChainTypeManager(outputPath, shouldSend);
+    }
```

Comment:
can we load it onchain?

Reply:
relying on a fact, that this was the latest deployment is wrong. `inputPathIfEcosystemDeployedLocally`